ch02/tacos/src/main/java/tacos/Taco.java

```diff
@@ -12,15 +12,17 @@
 public class Taco {
 
   // end::allButValidation[]
-  @NotNull
+  //@NotNull
+  @NotBlank(message="")
```

Comment:
Is your intention to avoid someone using spaces for example as a name?
If so you can still violate the minimum character limit if you use 5 spaces and 1 character.
I assume we were not validating the content of this field other than checking that the number of characters are at least 5. As such the original @NotNull suffices.